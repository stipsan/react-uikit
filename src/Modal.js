import classNames from 'classnames'
import React, { Component, PropTypes } from 'react'

import Button from './Button'

export default class Modal extends Component
{
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
  }

  state = { isActive: false }
  handleClick = () => {
    this.setState({ isActive: !this.state.isActive })
  }


  render() {
    const className = classNames('uk-modal', {
      'uk-open uk-display-block': this.state.isActive,
    })
    return (
      <div className="uk-overflow-container">
        <Button
          primary
          className="uk-button"
          onClick={this.handleClick}
        >
          Open
        </Button>
        <div className={className} aria-hidden={this.state.isActive}>
          <div className="uk-modal-dialog">
            <a
              className="uk-modal-close uk-close"
              onClick={this.handleClick}
            >
            </a>
            {this.props.children}
          </div>
        </div>
      </div>
    )
  }
}
