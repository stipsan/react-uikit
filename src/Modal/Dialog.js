import classNames from 'classnames'
import { Component, PropTypes, createElement } from 'react'

export default class Dialog extends Component {
  static defaultProps = {
    blank: false,
    caption: '',
    close: true,
    footer: [],
    header: '',
    large: false,
    lightbox: false,
  }

  static propTypes = {
    children: PropTypes.node.isRequired,
    handleClose: PropTypes.func.isRequired,
    blank: PropTypes.bool,
    caption: PropTypes.string,
    close: PropTypes.bool,
    footer: PropTypes.array, // eslint-disable-line react/forbid-prop-types
    header: PropTypes.string,
    large: PropTypes.bool,
    lightbox: PropTypes.bool,
  }

  handleOverlayClick = event => event.stopPropagation()

  handleClose = (callback) => {
    this.props.handleClose(callback)
  }

  render() {
    const { handleOverlayClick, handleClose } = this
    const {
      children,
      header,
      footer,
      caption,
      lightbox,
      blank,
      close,
      large,
     } = this.props

    const dialogClassName = classNames('uk-modal-dialog', {
      'uk-modal-dialog-lightbox': lightbox,
      'uk-modal-dialog-blank': blank,
      'uk-modal-dialog-large': large,
    })
    const closeClassName = classNames('uk-modal-close uk-close', {
      'uk-close-alt': lightbox,
    })

    return (
      <div className={dialogClassName} onClick={handleOverlayClick}>
        {close && (
          <a
            className={closeClassName}
            onClick={handleClose}
          />
        )}
        {header && (<div className="uk-modal-header"><h2>{header}</h2></div>)}
        {caption && (<div className="uk-modal-caption">{caption}</div>)}
        {children}
        { footer.length > 0 &&
          (<div className="uk-modal-footer uk-text-right">
            {
              footer.map((component, i) => createElement(component, { key: i, handleClose }))
            }
          </div>)
        }
      </div>
    )
  }
}
