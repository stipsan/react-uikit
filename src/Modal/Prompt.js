import { Component, PropTypes } from 'react'

export default class Prompt extends Component {

  static defaultProps = {
    isOpen: false,
  }

  static propTypes = {
    isOpen: PropTypes.bool,
  }

  componentDidUpdate(prevProps) {
    if (prevProps.isOpen === false && this.props.isOpen === true) {
      this.input.focus()
    }
  }

  render() {
    return (
      <div className="uk-margin-small-top uk-modal-content uk-form">
        <p>
          <input
            className="uk-width-1-1"
            ref={(node) => {
              if (node) {
                this.input = node
              }
            }}
            type="text"
          />
        </p>
      </div>
    )
  }
}
