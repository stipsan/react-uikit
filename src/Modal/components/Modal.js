import ExecutionEnvironment from 'exenv'
import React, { Component, PropTypes } from 'react'
// const ExecutionEnvironment = require('exenv')
const ModalPortal = React.createFactory(require('./ModalPortal'))
const ariaAppHider = require('../helpers/ariaAppHider')
const elementClass = require('element-class')
const renderSubtreeIntoContainer = require('react-dom').unstable_renderSubtreeIntoContainer
const Assign = require('lodash.assign')

const SafeHTMLElement = ExecutionEnvironment.canUseDOM ? window.HTMLElement : {}
const AppElement = ExecutionEnvironment.canUseDOM ? document.body : { appendChild() {} }

export default class Modal extends Component {

  displayName: 'Modal'

  statics = {
    setAppElement(element) {
      AppElement = ariaAppHider.setElement(element)
    },
    injectCSS() {
      'production' !== process.env.NODE_ENV
        && console.warn('React-Modal: injectCSS has been deprecated ' +
                        'and no longer has any effect. It will be removed in a later version')
    },
  }

  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    style: PropTypes.shape({
      content: PropTypes.object,
      overlay: PropTypes.object,
    }),
    portalClassName: PropTypes.string,
    appElement: PropTypes.instanceOf(SafeHTMLElement),
    onAfterOpen: PropTypes.func,
    onRequestClose: PropTypes.func,
    closeTimeoutMS: PropTypes.number,
    ariaHideApp: PropTypes.bool,
    shouldCloseOnOverlayClick: PropTypes.bool,
  }

  static defaultProps = () => {
    return {
      isOpen: false,
      portalClassName: 'ReactModalPortal',
      ariaHideApp: true,
      closeTimeoutMS: 0,
      shouldCloseOnOverlayClick: true,
    }
  }

  componentDidMount = () => {
    this.node = document.createElement('div')
    this.node.className = this.props.portalClassName
    document.body.appendChild(this.node)
    this.renderPortal(this.props)
  }

  componentWillReceiveProps = (newProps) => {
    this.renderPortal(newProps)
  }

  componentWillUnmount = () => {
    ReactDOM.unmountComponentAtNode(this.node)
    document.body.removeChild(this.node)
    elementClass(document.body).remove('ReactModal__Body--open')
  }

  renderPortal = (props) => {
    if (this.props.isOpen !== props.isOpen) {
      if (props.isOpen) {
        elementClass(document.body).add('ReactModal__Body--open')
      } else {
        elementClass(document.body).remove('ReactModal__Body--open')
      }

      if (props.ariaHideApp) {
        ariaAppHider.toggle(props.isOpen, props.appElement)
      }
    }

    this.portal = renderSubtreeIntoContainer(this, ModalPortal(Assign({}, props, { defaultStyles: Modal.defaultStyles })), this.node)
  }

  render() {
    return React.DOM.noscript()
  }
}

Modal.defaultStyles = {
  overlay: {
    position: 'fixed',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: 'rgba(255, 255, 255, 0.75)',
  },
  content: {
    position: 'absolute',
    top: '40px',
    left: '40px',
    right: '40px',
    bottom: '40px',
    border: '1px solid #ccc',
    background: '#fff',
    overflow: 'auto',
    WebkitOverflowScrolling: 'touch',
    borderRadius: '4px',
    outline: 'none',
    padding: '20px',
  },
}

// module.exports = Modal
