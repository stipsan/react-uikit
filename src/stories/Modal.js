import { storiesOf, action } from '@kadira/storybook'
import { Component } from 'react'
import { Button, Modal } from 'uikit-react'

Modal.displayName = 'Modal'

/* eslint-disable */
class OpenButton extends Component {
  handleClick = (event) => {
    action('handleOpen')()
    this.props.onClick(event)
  }
  render() {
    const { handleClick } = this
    return <Button onClick={handleClick}>Open</Button>
  }
}

class CancelButton extends Component {
  handleClick = () => {
    action('handleCancel')()
    this.props.handleClose()
  }
  render() {
    const { handleClick } = this
    return <Button onClick={handleClick}>Cancel</Button>
  }
}

class SaveButton extends Component {
  handleClick = () => {
    action('handleSave')()
    this.props.handleClose()
  }
  render() {
    const { handleClick } = this
    return <Button primary onClick={handleClick} className="uk-margin-small-left">Save</Button>
  }
}
/* eslint-enable */

const LoremIpsum = `Lorem ipsum dolor sit amet, consectetur adipisicing elit,
  sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
  Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
  commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
  dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
culpa qui officia deserunt mollit anim id est laborum.`

storiesOf('Modal', module)
  .addWithInfo('Basic Usage', '', () => (

    <div className="uk-margin-bottom">
      <Modal target={<OpenButton />}>
        <h1>Headline</h1>
        <p>{LoremIpsum}</p>
      </Modal>
    </div>
        ), { header: false, inline: true, propTables: [Modal] })
        .addWithInfo('Header and Footer', '', () => (
          <div className="uk-margin-bottom">
            <Modal
              footer={[CancelButton, SaveButton]}
              header="Headline"
              target={<OpenButton />}
            >
              <p>{LoremIpsum}</p>
            </Modal>
          </div>
  ), { header: false, inline: true })
  .addWithInfo('Caption', '', () => (
    <div className="uk-margin-bottom">
      <Modal caption="Caption" target={<OpenButton />}>
        <h1>Headline</h1>
        <p>{LoremIpsum}</p>
      </Modal>
    </div>
  ), { header: false, inline: true })
  .addWithInfo('Lightbox modifier', '', () => (
    <div className="uk-margin-bottom">
      <Modal lightbox target={<OpenButton />}>
        <img
          alt=""
          height="400"
          role="presentation"
          src="http://getuikit.com/docs/images/placeholder_600x400.svg"
          width="600"
        />
      </Modal>
    </div>
  ), { header: false, inline: true })
  .addWithInfo('Blank', '', () => (
    <div className="uk-margin-bottom">
      <Modal blank target={<OpenButton />} >
        <div className="uk-grid uk-flex-middle">
          <div
            className="uk-width-medium-1-2 uk-height-viewport uk-cover-background uk-row-first"
            style={{
              backgroundImage: 'url(http://getuikit.com/docs/images/placeholder_120x255.svg)',
            }}
          />
          <div className="uk-width-medium-1-2">
            <h1>Headline</h1>
            <div className="uk-width-medium-1-3">
              <p>{LoremIpsum}</p>
            </div>
          </div>
        </div>
      </Modal>
    </div>
  ), { header: false, inline: true })
  .addWithInfo('Spinner', '', () => (
    <div className="uk-margin-bottom">
      <Modal
        close={false}
        target={<OpenButton />}
      >
        <div className="uk-modal-spinner" />
      </Modal>
    </div>
  ), { header: false, inline: true })
  /*
  .addWithInfo('Center', '', () => (
    <div className="uk-margin-bottom">
      <Modal center target={OpenButton} >
        <h1>Headline</h1>
        <p>{LoremIpsum}</p>
      </Modal>
    </div>
  ), { header: false, inline: true })
  // */
  .addWithInfo('Large dialog modifier', '', () => (
    <div className="uk-margin-bottom">
      <Modal large target={<OpenButton />} >
        <h1>Headline</h1>
        <p>{LoremIpsum}</p>
      </Modal>
    </div>
  ), { header: false, inline: true })
  /*
  .addWithInfo('Overflow container in modal', '', () => (
    <div className="uk-margin-bottom">
      <Modal target={OpenButton}>
        <h1>Headline</h1>
        <h2>Some text above the scrollable box</h2>
        <p>{LoremIpsum}</p>
        <div className="uk-overflow-container">
          <h2>Overflow container</h2>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
          <p>{LoremIpsum}</p>
        </div>
        <p>{LoremIpsum}</p>
      </Modal>
    </div>
  ), { header: false, inline: true })
  // */
  .addWithInfo('Alert', '', () => (
    <div className="uk-margin-bottom">
      <Modal
        // eslint-disable-next-line react/prop-types,react/jsx-no-bind
        target={<Button>Alert</Button>}
        type="alert"
        onConfirm={action('onConfirm')}
      >
        Attention!
      </Modal>
    </div>
  ), { header: false, inline: true })
  .addWithInfo('Confirm', '', () => (
    <div className="uk-margin-bottom">
      <Modal
        cancelButtonClass="uk-button-primary"
        confirmButtonClass="uk-button-danger"
        confirmLabel="Delete"
        // eslint-disable-next-line react/prop-types,react/jsx-no-bind
        target={<Button>Confirm</Button>}
        type="confirm"
        onCancel={action('onCancel')}
        onConfirm={action('onConfirm')}
      >
        Are you sure?
      </Modal>
    </div>
  ), { header: false, inline: true })
  /*
  .addWithInfo('Prompt', '', () => (
    <div className="uk-margin-bottom">
      <Modal
        // eslint-disable-next-line react/prop-types,react/jsx-no-bind
        target={({ handleOpen }) => <Button onClick={handleOpen}>Prompt</Button>}
        type="prompt"
        onBlur={action('onBlur')}
        onCancel={action('onCancel')}
        onChange={action('onChange')}
        onConfirm={action('onConfirm')}
        onFocus={action('onFocus')}
      >
        Name:
      </Modal>
    </div>
  ), { header: false, inline: true })
  .addWithInfo('Block UI', '', () => (
    <div className="uk-margin-bottom">
      <Modal
        // eslint-disable-next-line react/prop-types,react/jsx-no-bind
        target={({ handleOpen }) => <Button onClick={handleOpen}>Block UI</Button>}
        type="blockUI"
        // eslint-disable-next-line react/jsx-no-bind
        onOpen={close => setTimeout(() => close(), 5000)}
      >
        Wait 5 sec...
      </Modal>
    </div>
  ), { header: false, inline: true })
  //*/
