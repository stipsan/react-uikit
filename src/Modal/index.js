import classNames from 'classnames'
import KeyHandler, { KEYDOWN } from 'react-key-handler'
import { Component, PropTypes, createElement } from 'react'

import Button from '../Button'
import Dialog from './Dialog'
import Prompt from './Prompt'

export default class Modal extends Component {

  static propTypes = {
    cancelLabel: PropTypes.string,
    children: PropTypes.node,
    className: PropTypes.string,
    confirmLabel: PropTypes.string,
    isOpen: PropTypes.bool,
    target: PropTypes.func,
    type: PropTypes.string,
    onCancel: PropTypes.func,
    onConfirm: PropTypes.func,
  }

  static defaultProps = {
    isOpen: false,
    closeTimeoutMS: 0,
    shouldCloseOnOverlayClick: true,
    target: 'button',
    lightbox: false,
    type: 'default',
    confirmLabel: 'Ok',
    cancelLabel: 'Cancel',
  }

  state = { shouldDisplay: false, isOpen: false }


  handleClick = () => {
    this.setState({ shouldDisplay: !this.state.shouldDisplay })
  }

  handleOpen = () => {
    this.setState(
      { shouldDisplay: true },
      () => setTimeout(() => this.setState({ isOpen: true }), 50)
    )
  }

  handleClose = () => {
    this.setState(
      { isOpen: false },
      () => setTimeout(() => this.setState({ shouldDisplay: false }), 300)
    )
  }

  handleConfirm = () => {
    this.props.onConfirm()
    this.handleClose()
  }

  handleCancel = () => {
    this.props.onCancel()
    this.handleClose()
  }

  render() {
    const {
      handleClose, handleOpen, handleConfirm, handleCancel,
     } = this
    const {
      isOpen,
      type,
      confirmLabel,
      cancelLabel,
    } = this.props
    const target = this.props.target &&
          createElement(this.props.target, { handleOpen, children: 'Open' })
    const className = classNames('uk-modal', {
      'uk-open': this.state.shouldDisplay && this.state.isOpen,
    })
    const style = {
      display: this.state.shouldDisplay && 'block',
    }

    let footer = []
    if (type === 'alert') {
      footer = [() => <Button primary onClick={handleConfirm}>{confirmLabel}</Button>]
    }

    if (type === 'confirm' || type === 'prompt') {
      footer = [
        () => <Button onClick={handleCancel}>{cancelLabel}</Button>,
        () => <span>&nbsp;</span>,
        () => <Button primary onClick={handleConfirm}>{confirmLabel}</Button>,
      ]
    }
    const { children, ...dialogProps } = this.props

    let dialogChildren
    if (type === 'prompt') {
      dialogChildren = <Prompt isOpen={this.state.isOpen}>{children}</Prompt>
    } else {
      dialogChildren = children
    }

    return (
      <div>
        <KeyHandler keyEventName={KEYDOWN} keyValue="Escape" onKeyHandle={handleClose} />
        {target}
        <div
          aria-hidden={isOpen}
          className={className}
          style={style}
          onClick={type === 'default' && handleClose}
        >
          <Dialog
            close={type === 'default'}
            footer={footer}
            handleClose={handleClose}
            {...dialogProps}
          >
            {dialogChildren}
          </Dialog>
        </div>
      </div>
    )
  }
}
