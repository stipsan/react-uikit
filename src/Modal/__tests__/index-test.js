import renderer from 'react-test-renderer'

import Modal from '../index'

describe('Modal', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Modal>Lorem ipsum</Modal>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders a Header and Footer', () => {
    const component = renderer.create(
      <Modal
        footer={[
          // eslint-disable-next-line react/prop-types
          ({ handleClose }) => <button onClick={handleClose}>Yes!</button>,
        ]}
        header="Headline"
      >
        Lorem ipsum
      </Modal>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders a Caption', () => {
    const component = renderer.create(
      <Modal
        caption="Caption"
      >
        Lorem ipsum
      </Modal>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders a Lightbox', () => {
    const component = renderer.create(
      <Modal
        lightbox
      >
        <img role="presentation" src="http://getuikit.com/docs/images/placeholder_600x400.svg" />
      </Modal>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders a Blank Modal', () => {
    const component = renderer.create(
      <Modal
        blank
      >
        <div className="uk-grid uk-flex-middle">
          <div
            className="uk-width-medium-1-2 uk-height-viewport uk-cover-background uk-row-first"
            style={{
              backgroundImage: 'url(http://getuikit.com/docs/images/placeholder_120x255.svg)',
            }}
          />
          <div className="uk-width-medium-1-2">
            <h1>Headline</h1>
            <div className="uk-width-medium-1-3">
              <p>Lorem ipsum</p>
            </div>
          </div>
        </div>
      </Modal>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders a Spinner', () => {
    const component = renderer.create(
      <Modal
        close={false}
      >
        <div className="uk-modal-spinner" />
      </Modal>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders an alert modal', () => {
    const component = renderer.create(
      <Modal
        type="alert"
      >
        Something bad happened!
      </Modal>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
