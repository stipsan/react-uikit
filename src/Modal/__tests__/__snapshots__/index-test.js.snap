// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders a Blank Modal 1`] = `
<Portal
  beforeClose={[Function]}
  closeOnEsc={true}
  closeOnOutsideClick={true}
  onClose={[Function]}
  onOpen={[Function]}
  openByClickOn={
    <button>
      Open
    </button>
  }
>
  <div
    aria-hidden={false}
    className="uk-modal"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="uk-modal-dialog uk-modal-dialog-blank"
      onClick={[Function]}
    >
      <a
        className="uk-modal-close uk-close"
        onClick={[Function]}
      />
      
      
      <div
        className="uk-modal-body"
      >
        <div
          className="uk-grid uk-flex-middle"
        >
          <div
            className="uk-width-medium-1-2 uk-height-viewport uk-cover-background uk-row-first"
            style={
              Object {
                "backgroundImage": "url(http://getuikit.com/docs/images/placeholder_120x255.svg)",
              }
            }
          />
          <div
            className="uk-width-medium-1-2"
          >
            <h1>
              Headline
            </h1>
            <div
              className="uk-width-medium-1-3"
            >
              <p>
                Lorem ipsum
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</Portal>
`;

exports[`Modal renders a Caption 1`] = `
<Portal
  beforeClose={[Function]}
  closeOnEsc={true}
  closeOnOutsideClick={true}
  onClose={[Function]}
  onOpen={[Function]}
  openByClickOn={
    <button>
      Open
    </button>
  }
>
  <div
    aria-hidden={false}
    className="uk-modal"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="uk-modal-dialog"
      onClick={[Function]}
    >
      <a
        className="uk-modal-close uk-close"
        onClick={[Function]}
      />
      
      <div
        className="uk-modal-caption"
      >
        Caption
      </div>
      <div
        className="uk-modal-body"
      >
        Lorem ipsum
      </div>
    </div>
  </div>
</Portal>
`;

exports[`Modal renders a Header and Footer 1`] = `
<Portal
  beforeClose={[Function]}
  closeOnEsc={true}
  closeOnOutsideClick={true}
  onClose={[Function]}
  onOpen={[Function]}
  openByClickOn={
    <button>
      Open
    </button>
  }
>
  <div
    aria-hidden={false}
    className="uk-modal"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="uk-modal-dialog"
      onClick={[Function]}
    >
      <a
        className="uk-modal-close uk-close"
        onClick={[Function]}
      />
      <div
        className="uk-modal-header"
      >
        <h2>
          Headline
        </h2>
      </div>
      
      <div
        className="uk-modal-body"
      >
        Lorem ipsum
      </div>
      <div
        className="uk-modal-footer uk-text-right"
      >
        <button
          onClick={[Function]}
        >
          Yes!
        </button>
      </div>
    </div>
  </div>
</Portal>
`;

exports[`Modal renders a Lightbox 1`] = `
<Portal
  beforeClose={[Function]}
  closeOnEsc={true}
  closeOnOutsideClick={true}
  onClose={[Function]}
  onOpen={[Function]}
  openByClickOn={
    <button>
      Open
    </button>
  }
>
  <div
    aria-hidden={false}
    className="uk-modal"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="uk-modal-dialog uk-modal-dialog-lightbox"
      onClick={[Function]}
    >
      <a
        className="uk-modal-close uk-close uk-close-alt"
        onClick={[Function]}
      />
      
      
      <div
        className="uk-modal-body"
      >
        <img
          alt=""
          role="presentation"
          src="http://getuikit.com/docs/images/placeholder_600x400.svg"
        />
      </div>
    </div>
  </div>
</Portal>
`;

exports[`Modal renders a Spinner 1`] = `
<Portal
  beforeClose={[Function]}
  closeOnEsc={true}
  closeOnOutsideClick={true}
  onClose={[Function]}
  onOpen={[Function]}
  openByClickOn={
    <button>
      Open
    </button>
  }
>
  <div
    aria-hidden={false}
    className="uk-modal"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="uk-modal-dialog"
      onClick={[Function]}
    >
      
      
      <div
        className="uk-modal-body"
      >
        <div
          className="uk-modal-spinner"
        />
      </div>
    </div>
  </div>
</Portal>
`;

exports[`Modal renders an alert modal 1`] = `
<Portal
  beforeClose={[Function]}
  closeOnEsc={true}
  closeOnOutsideClick={true}
  onClose={[Function]}
  onOpen={[Function]}
  openByClickOn={
    <button>
      Open
    </button>
  }
>
  <div
    aria-hidden={false}
    className="uk-modal"
    onClick={false}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="uk-modal-dialog"
      onClick={[Function]}
    >
      
      
      <div
        className="uk-modal-body"
      >
        Something bad happened!
      </div>
      <div
        className="uk-modal-footer uk-text-right"
      >
        <button
          className="uk-button uk-button-primary"
          onClick={[Function]}
          type="button"
        >
          Ok
        </button>
      </div>
    </div>
  </div>
</Portal>
`;

exports[`Modal renders an confirm modal 1`] = `
<Portal
  beforeClose={[Function]}
  closeOnEsc={true}
  closeOnOutsideClick={true}
  onClose={[Function]}
  onOpen={[Function]}
  openByClickOn={
    <button>
      Open
    </button>
  }
>
  <div
    aria-hidden={false}
    className="uk-modal"
    onClick={false}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="uk-modal-dialog"
      onClick={[Function]}
    >
      
      
      <div
        className="uk-modal-body"
      >
        Are you absolutely sure about this?
      </div>
      <div
        className="uk-modal-footer uk-text-right"
      >
        <button
          className="uk-button uk-button-default"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <span>
           
        </span>
        <button
          className="uk-button uk-button-primary"
          onClick={[Function]}
          type="button"
        >
          Ok
        </button>
      </div>
    </div>
  </div>
</Portal>
`;

exports[`Modal renders an prompt modal 1`] = `
<Portal
  beforeClose={[Function]}
  closeOnEsc={true}
  closeOnOutsideClick={true}
  onClose={[Function]}
  onOpen={[Function]}
  openByClickOn={
    <button>
      Open
    </button>
  }
>
  <div
    aria-hidden={false}
    className="uk-modal"
    onClick={false}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="uk-modal-dialog"
      onClick={[Function]}
    >
      
      
      <div
        className="uk-modal-body"
      >
        <div
          className="uk-margin-small-top uk-modal-content uk-form"
        >
          <p>
            <input
              className="uk-width-1-1"
              type="text"
            />
          </p>
        </div>
      </div>
      <div
        className="uk-modal-footer uk-text-right"
      >
        <button
          className="uk-button uk-button-default"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <span>
           
        </span>
        <button
          className="uk-button uk-button-primary"
          onClick={[Function]}
          type="button"
        >
          Ok
        </button>
      </div>
    </div>
  </div>
</Portal>
`;

exports[`Modal renders correctly 1`] = `
<Portal
  beforeClose={[Function]}
  closeOnEsc={true}
  closeOnOutsideClick={true}
  onClose={[Function]}
  onOpen={[Function]}
  openByClickOn={
    <button>
      Open
    </button>
  }
>
  <div
    aria-hidden={false}
    className="uk-modal"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="uk-modal-dialog"
      onClick={[Function]}
    >
      <a
        className="uk-modal-close uk-close"
        onClick={[Function]}
      />
      
      
      <div
        className="uk-modal-body"
      >
        Lorem ipsum
      </div>
    </div>
  </div>
</Portal>
`;
